documentation/jetty/modules/programming-guide/pages/arch/threads.adoc

```diff
@@ -115,48 +115,45 @@ This mode solves the head-of-line blocking discussed in the <<execution-strategy
+In the `Execute-Produce-Consume` mode, there is no dedicated producer thread. Rather a `Thread x` is initially the producer and when it has produced a task (`Pn`) it submits an internal task to an `Executor` so `Thread y` can take over production (`Pn+1`), and the original `Thread x` directly runs the task it produced (`Rn`).
 
 [plantuml]
 ----
 skinparam backgroundColor transparent
 
-compact concise "Thread 1" as WT1
-compact concise "Thread 2" as WT2
-compact concise "Thread 3" as WT3
-compact concise "Thread 4" as WT4
+compact concise "Thread A" as WT1
+compact concise "Thread B" as WT2
+compact concise "Thread C" as WT3
+compact concise "Thread D" as WT4
 hide time-axis
 
 @WT1
-0 is T1 #lightgreen
-1 is "Run T1" #dodgerblue
+0 is P1 #lightgreen
+1 is R1 #dodgerblue
 5 is {hidden}
 
 @WT2
-1 is T2 #lightgreen
-2 is "Run T2" #dodgerblue
-4 is T5 #lightgreen
-5 is "Run T5" #dodgerblue
+1 is P2 #lightgreen
+2 is R2 #dodgerblue
+4 is P5 #lightgreen
+5 is R5 #dodgerblue
 10 is {hidden}
 
 @WT3
-2 is T3 #lightgreen
-3 is "Run T3" #dodgerblue
+2 is P3 #lightgreen
+3 is R3 #dodgerblue
 6 is {hidden}
 
 @WT4
-3 is T4 #lightgreen
-4 is "Run T4" #dodgerblue
+3 is P4 #lightgreen
+4 is R4 #dodgerblue
 8 is {hidden}
 
 ----
 
-This mode may operate like <<execution-strategy-pc,`Produce-Consume`>> when the take-over production task run, for example, by thread `Thread 3` takes time to be executed (for example, in a busy server): then thread `Thread 2` will produce one task and run it, then produce another task and run it, etc. -- `Thread 2` behaves exactly like the `Produce-Consume` mode.
-By the time thread `Thread 3` takes over task production from `Thread 2`, all the work might already be done.
+This mode may operate similar to <<execution-strategy-pc,`Produce-Consume`>> as after running a task (`Rn`) a thread may try to become a producer again (`Pn+m`), as `Thread B` does above.
 
-This mode may also operate similarly to <<execution-strategy-pec,`Produce-Execute-Consume`>> when the take-over production task always finds a free CPU core immediately (for example, in a mostly idle server): thread `Thread 1` will produce a task, yield production to `Thread 2` while `Thread 1` is running the task; `Thread 2` will produce a task, yield production to `Thread 3` while `Thread 2` is running the task, etc.
-
-Differently from `Produce-Execute-Consume`, here production happens on different threads, but the advantage is that the task is run by the same thread that produced it (which is CPU core cache efficient).
+Differently from `Produce-Execute-Consume`, here production happens on different threads, but the advantage is that the task is run by the same thread that produced it (which is CPU core cache efficient). However, if the executor does not have an idle thread immediately available then using this strategy may result in production being queued waiting for an available thread, which can result in dead lock.
```

Comment:
`which can result in dead lock` is too vague, or assuming too much context: there may not be any two threads mutually waiting explicitly for each other's already held lock.

I'd rather speak of a hang rather than a deadlock in this case.